submission/pipelines/generate_pipelines.py

```diff
@@ -400,29 +402,71 @@ def remove_digests(
     return pipeline_json_structure
 
 
+def update_digest(
+    pipeline_json_structure, filename=None
+):
+    """
+    This function updates the pipeline's digests and version numbers
+
+    Parameters
+    ----------
+    pipeline_json_structure: the pipeline in JSON form (WITHOUT) digests.  This or the `filename` parameter is mandatory
+    filename: the filename of the pipeline json, so we can read it in
+
+    :return a pipeline with updated digests
+    """
+    if pipeline_json_structure is None and filename is None:
+        raise Exception
+    elif pipeline_json_structure is None:
+        with open(filename, "r") as file:
+            # NOTE: must be a pipeline with no digests, or recent digests
+            # NOTE: reading this in as straight JSON doesn't work so we have to use the pipeline_module
+            pipeline_to_run = pipeline_module.Pipeline.from_json(string_or_file=file).to_json_structure()
+    else:
+        pipeline_to_run = pipeline_module.Pipeline.from_json(json.dumps(pipeline_json_structure)).to_json_structure()
+
+    for step in pipeline_to_run['steps']:
+        # if not updated, check and update
+        primitive = pipeline_module.PrimitiveStep(
+            primitive=d3m_index.get_primitive(
+                step["primitive"]["python_path"]
+            )
+        )
+        check_step = primitive.to_json_structure()
+        # lets verify that both are updated
+        assert check_step["primitive"]["version"] == step["primitive"]["version"], "Updating version failed"
+        assert check_step["primitive"]["digest"] == step["primitive"]["digest"], "Updating digest failed"
```

Comment:
The only digest in the whole pipeline should be the digest of the primitive it is demonstrating, i.e. our primitives.

Reply:
Yeah, we don't add the digest, so I'll remove this line